In [13]:
!git clone https://github.com/hafizahmadadilaiengineer/flyrank-ml-internship.git

fatal: destination path 'flyrank-ml-internship' already exists and is not an empty directory.


In [14]:
!ls flyrank-ml-internship/data/raw

content_refresh_anonymized.csv


# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## Selected Lane

I selected **Lane 2: Refresh / Content Opportunity Scoring**.

This lane focuses on helping content and SEO teams prioritize which pages should be reviewed first for content refresh or optimization. Instead of replacing human decision-making, the goal is to build a decision-support system that ranks pages using observable search and engagement signals.

I selected this lane because it combines exploratory data analysis, feature engineering, machine learning, and ranking evaluation while solving a practical business problem.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question

Which content pages should be reviewed first because they show evidence of declining performance while still receiving meaningful search visibility?

### Decision

Prioritize which pages should enter the content review queue first.

### Who Acts?

SEO specialists and content editors.

### Action

The highest-ranked pages will be reviewed for possible content updates, metadata improvements, expansion, or monitoring.

### Cost of a Wrong Recommendation

If a low-value page is ranked too high, editors waste valuable time reviewing pages with little opportunity.

If a high-value page is ranked too low, important content opportunities may be missed, potentially resulting in continued traffic decline.

### Why Machine Learning?

Machine learning can combine multiple observable search and engagement signals simultaneously, allowing it to prioritize pages more effectively than simple rule-based approaches.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
from pathlib import Path
import pandas as pd

repo_root = Path("flyrank-ml-internship")
df = pd.read_csv(repo_root / "data/raw/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)
print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", df["ctr"].mean())
print("Average Impressions:", df["impressions_90d"].mean())
print("Average Sessions:", df["sessions_90d"].mean())

Dataset Shape: (30000, 44)

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.5107333333333334
Average Impressions: 5200.3663
Average Sessions: 37.066633333333336


The starter dataset contains **30,000 content pages** with **44 features**, providing sufficient data for exploratory analysis and model development.

Among these pages, **16,262 are labeled as "down"**, indicating that declining content represents a significant portion of the dataset and provides enough examples for studying content refresh opportunities.

The dataset also has an **average of 5,200 impressions over the last 90 days** and an **average CTR of approximately 0.51%**, showing that many pages receive measurable search visibility while still leaving room for improvement. These observations suggest that ranking pages for content review is a meaningful and practical machine learning problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project identifies patterns observed in historical search and engagement data.

The results will be used as decision support rather than proof of causal relationships.

The recommendations suggest which pages may deserve human review based on observable evidence.

This work does not prove how Google's ranking systems operate, nor does it guarantee that refreshing a page will improve performance.

- [x] Every section above is filled.
- [x] Notebook runs from top to bottom with no errors.
- [x] No client names, URLs, or private queries are included.
- [x] My claims are careful and evidence-based.
- [x] Notebook will be committed under work/notebooks/.